src/posts/posts.repository.ts

```diff
@@ -39,8 +46,8 @@ export class PostsRepository extends Repository<PostSocial> {
     return found;
   }
 
-  async deletePost(id: string): Promise<string> {
-    const found = await this.getPostById(id);
+  async deletePost(id: string, user: User): Promise<string> {
+    const found = await this.getPostById(id, user);
```

Comment:
Đặt tên biến cụ thể hơn e nhé
```suggestion
    const post = await this.getPostById(id, user);
```